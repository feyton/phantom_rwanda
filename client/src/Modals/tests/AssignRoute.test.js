import { fireEvent, render, screen } from '@testing-library/react';
import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import AssignRouteModal, { assignRouteToBus } from '../AssignRoute.js';

const mockUseLocationValue = {
  pathname: '/testroute',
  search: '',
  hash: '',
  state: {
    plateNumber: 'RAB162H',
    busType: 'coaster',
    seats: 25,
    route: {
      origin: 'Nyabugogo',
      destination: 'Remera'
    }
  }
};
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockImplementation(() => {
    return mockUseLocationValue;
  })
}));

describe('Modal tests', () => {
  const event = {
    target: {
      route: { value: 'some value' }
    }
  };
  it('It should test the onSubmit function', () => {
    const onSubmit = jest.fn();
    const component = mount(
      <MemoryRouter>
        <AssignRouteModal onSubmit={onSubmit()} />
      </MemoryRouter>
    );
    const form = component.find('form');
    form.simulate('submit', event);
    expect(onSubmit).toBeCalledTimes(1);
  });
  it('Should render the assign modal', async () => {
    const onSubmitMock = jest.fn();
    const handleCloseMock = jest.fn();

    const elem = render(
      <MemoryRouter>
        <AssignRouteModal
          onSubmit={onSubmitMock}
          handleClose={handleCloseMock}
        />
      </MemoryRouter>
    );

    fireEvent.change(screen.getByTestId('bus-route'), {
      target: {
        value: 'RN45'
      }
    });

    fireEvent.submit(screen.getByTestId('assign-form'), {
      target: {
        route: { value: 'some value' }
      }
    });

    expect(elem).toMatchSnapshot();
  });
});
