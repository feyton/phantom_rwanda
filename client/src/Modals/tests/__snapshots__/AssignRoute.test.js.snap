// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests Should render the assign modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="swal2-shown swal2-height-auto"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="font-raleway w-fit px-8 py-2"
      >
        <h1
          class="font-bold"
        >
          Bus Info
        </h1>
        <hr />
        <ul>
          <li>
            <span>
              Plate: 
            </span>
            <b>
              RAB162H
            </b>
          </li>
          <li>
            <span>
              Type: 
            </span>
            <b>
              coaster
            </b>
          </li>
          <li>
            <span>
              Seats: 
            </span>
            <b>
              25
            </b>
          </li>
          <li>
            <span>
              Assigned route: 
            </span>
            <b>
              Nyabugogo
               - 
              Remera
            </b>
          </li>
        </ul>
        <hr />
        <div>
          <form
            action="#23"
            class="py-2"
            id="assign-form"
          >
            <label
              class="block font-raleway"
              for="bus"
            >
              Route:
               
            </label>
            <select
              class="bg-gray-300 my-2 border px-2 py-1 rounded-sm placeholder:font-raleway uppercase"
              id="bus-route"
              name="route"
              required=""
              type="text"
            >
              <option
                id="origin-select"
              >
                Select route
              </option>
            </select>
            <button
              class="bg-primary rounded-md text-white font-bold font-raleway py-2 px-3 hover:bg-hover transition-all block hover:transition-all"
              type="submit"
            >
              Assign
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      class="swal2-container swal2-center swal2-backdrop-show"
    >
      <div
        aria-describedby="swal2-html-container"
        aria-labelledby="swal2-title"
        aria-live="assertive"
        aria-modal="true"
        class="swal2-popup swal2-modal swal2-icon-warning"
        role="dialog"
        style="opacity: 0 !important; display: grid;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="swal2-close"
          style="display: none;"
          type="button"
        >
          Ã—
        </button>
        <ul
          class="swal2-progress-steps"
          style="display: none;"
        />
        <div
          class="swal2-icon swal2-warning swal2-icon-show"
          style="display: flex;"
        >
          <div
            class="swal2-icon-content"
          >
            !
          </div>
        </div>
        <img
          class="swal2-image"
          style="display: none;"
        />
        <h2
          class="swal2-title"
          id="swal2-title"
          style="display: block;"
        >
          Are you sure?
        </h2>
        <div
          class="swal2-html-container"
          id="swal2-html-container"
          style="display: block;"
        >
          This bus already have a route!
        </div>
        <input
          class="swal2-input"
          style="display: none;"
        />
        <input
          class="swal2-file"
          style="display: none;"
          type="file"
        />
        <div
          class="swal2-range"
          style="display: none;"
        >
          <input
            type="range"
          />
          <output />
        </div>
        <select
          class="swal2-select"
          style="display: none;"
        />
        <div
          class="swal2-radio"
          style="display: none;"
        />
        <label
          class="swal2-checkbox"
          for="swal2-checkbox"
          style="display: none;"
        >
          <input
            type="checkbox"
          />
          <span
            class="swal2-label"
          />
        </label>
        <textarea
          class="swal2-textarea"
          style="display: none;"
        />
        <div
          class="swal2-validation-message"
          id="swal2-validation-message"
          style="display: none;"
        />
        <div
          class="swal2-actions"
          style="display: flex;"
        >
          <div
            class="swal2-loader"
          />
          <button
            aria-label=""
            class="swal2-confirm swal2-styled swal2-default-outline"
            style="display: inline-block; background-color: rgb(48, 133, 214);"
            type="button"
          >
            Yes, Assign route!
          </button>
          <button
            aria-label=""
            class="swal2-deny swal2-styled"
            style="display: none;"
            type="button"
          >
            No
          </button>
          <button
            aria-label=""
            class="swal2-cancel swal2-styled swal2-default-outline"
            style="display: inline-block; background-color: rgb(221, 51, 51);"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div
          class="swal2-footer"
          style="display: none;"
        />
        <div
          class="swal2-timer-progress-bar-container"
        >
          <div
            class="swal2-timer-progress-bar"
            style="display: none;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="font-raleway w-fit px-8 py-2"
    >
      <h1
        class="font-bold"
      >
        Bus Info
      </h1>
      <hr />
      <ul>
        <li>
          <span>
            Plate: 
          </span>
          <b>
            RAB162H
          </b>
        </li>
        <li>
          <span>
            Type: 
          </span>
          <b>
            coaster
          </b>
        </li>
        <li>
          <span>
            Seats: 
          </span>
          <b>
            25
          </b>
        </li>
        <li>
          <span>
            Assigned route: 
          </span>
          <b>
            Nyabugogo
             - 
            Remera
          </b>
        </li>
      </ul>
      <hr />
      <div>
        <form
          action="#23"
          class="py-2"
          id="assign-form"
        >
          <label
            class="block font-raleway"
            for="bus"
          >
            Route:
             
          </label>
          <select
            class="bg-gray-300 my-2 border px-2 py-1 rounded-sm placeholder:font-raleway uppercase"
            id="bus-route"
            name="route"
            required=""
            type="text"
          >
            <option
              id="origin-select"
            >
              Select route
            </option>
          </select>
          <button
            class="bg-primary rounded-md text-white font-bold font-raleway py-2 px-3 hover:bg-hover transition-all block hover:transition-all"
            type="submit"
          >
            Assign
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
